diff -ura --color facefusion-3.6.0/facefusion/content_analyser.py facefusion-3.6.0.new/facefusion/content_analyser.py
--- facefusion-3.6.0/facefusion/content_analyser.py	2026-03-16 22:08:43.000000000 +0800
+++ facefusion-3.6.0.new/facefusion/content_analyser.py	2026-04-05 12:52:40.906941812 +0800
@@ -190,33 +190,7 @@
 
 
 def detect_nsfw(vision_frame : VisionFrame) -> bool:
-	is_nsfw_1 = detect_with_nsfw_1(vision_frame)
-	is_nsfw_2 = detect_with_nsfw_2(vision_frame)
-	is_nsfw_3 = detect_with_nsfw_3(vision_frame)
-
-	return is_nsfw_1 and is_nsfw_2 or is_nsfw_1 and is_nsfw_3 or is_nsfw_2 and is_nsfw_3
-
-
-def detect_with_nsfw_1(vision_frame : VisionFrame) -> bool:
-	detect_vision_frame = prepare_detect_frame(vision_frame, 'nsfw_1')
-	detection = forward_nsfw(detect_vision_frame, 'nsfw_1')
-	detection_score = numpy.max(numpy.amax(detection[:, 4:], axis = 1))
-	return bool(detection_score > 0.2)
-
-
-def detect_with_nsfw_2(vision_frame : VisionFrame) -> bool:
-	detect_vision_frame = prepare_detect_frame(vision_frame, 'nsfw_2')
-	detection = forward_nsfw(detect_vision_frame, 'nsfw_2')
-	detection_score = detection[0] - detection[1]
-	return bool(detection_score > 0.25)
-
-
-def detect_with_nsfw_3(vision_frame : VisionFrame) -> bool:
-	detect_vision_frame = prepare_detect_frame(vision_frame, 'nsfw_3')
-	detection = forward_nsfw(detect_vision_frame, 'nsfw_3')
-	detection_score = (detection[2] + detection[3]) - (detection[0] + detection[1])
-	return bool(detection_score > 10.5)
-
+	return False
 
 def forward_nsfw(vision_frame : VisionFrame, model_name : str) -> Detection:
 	content_analyser = get_inference_pool().get(model_name)
diff -ura --color facefusion-3.6.0/facefusion/core.py facefusion-3.6.0.new/facefusion/core.py
--- facefusion-3.6.0/facefusion/core.py	2026-03-16 22:08:43.000000000 +0800
+++ facefusion-3.6.0.new/facefusion/core.py	2026-04-05 12:50:27.136189233 +0800
@@ -118,10 +118,7 @@
 		voice_extractor
 	]
 
-	content_analyser_content = inspect.getsource(content_analyser).encode()
-	content_analyser_hash = hash_helper.create_hash(content_analyser_content)
-
-	return all(module.pre_check() for module in common_modules) and content_analyser_hash == 'b14e7b92'
+	return all(module.pre_check() for module in common_modules)
 
 
 def processors_pre_check() -> bool:
